File: problems/rice/Prob07.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package problems.rice;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;
import problems.Input;

/**
 *
 * @author rootie
 */
public class Prob07 {

    static class Ship {

        public int x;
        public int y;
        public int speed;
        public String name;

        public Ship(String info) {
            this.name = info.substring(0, info.indexOf("_"));
            switch (info.substring(info.indexOf("_") + 1, info.indexOf(":"))) {
                case "A":
                    this.speed = 10;
                    break;
                case "B":
                    this.speed = 20;
                    break;
                case "C":
                    this.speed = 30;
                    break;
            }
            this.x = Integer.parseInt(info.split(":")[1].split(",")[0]);
            this.y = Integer.parseInt(info.split(":")[1].split(",")[1]);
        }

        public void stepShip() {
            x -= speed;
        }

        public void destroyShip() {
            System.out.println("Destroyed Ship: " + name + " xLoc: " + x);
        }

    }

    public static void main(String[] args) throws FileNotFoundException {
        // get file obj
        Scanner scanner = new Scanner(new File(Input.JPATH + "Prob07.in.txt"));
        int testCaseAmt = Integer.parseInt(scanner.nextLine());
        for (int tCa = 0; tCa < testCaseAmt; tCa++) {
            ArrayList<Ship> ships = new ArrayList<>();
            // load ships from file
            int subTestCaseAmt = Integer.parseInt(scanner.nextLine());
            for (int i = 0; i < subTestCaseAmt; i++) {
                String data = scanner.nextLine();
                ships.add(new Ship(data));
            }
            // run sim with ships
            while (ships.size() > 1) {
                Ship target = new Ship("BOGUSSHIP_A:-1,-1");
                target.x = Integer.MAX_VALUE;
                // find closest ship
                for (Ship ship : ships) {
                    if (ship.x < target.x) {
                        target = ship;
                    } else if (ship.x == target.x) {
                        target = (ship.y > target.y) ? ship : target;
                    }
                }
                // destroy ship
                target.destroyShip();
                ships.remove(target);
                // step all ships
                for (Ship ship : ships) {
                    ship.stepShip();
                }
            }
            ships.get(0).destroyShip();
        }
        scanner.close();
    }

}

File: problems/rice_py/prob07.py
import path

path = path.j_input + "Prob07.in.txt"
file = open(path, "r")
amt = int(file.readline())

# ship data Touple format (name, x, y)
for i in range(0, amt):
	s_amt = int(file.readline())
	ships = []
	# Read ships
	for i2 in range(0, s_amt):
		line = file.readline()[0:-1]
		cords = line.split(":")[1]
		ships.append((line.split(":")[0], int(cords.split(",")[0]), int(cords.split(",")[1])))
	# Do Sim 
	while(len(ships) != 0):
		# distroy ship
		s = min(ships, key = lambda x: abs(x[1]) - abs(x[2]))
		(d_name, d_x, d_y) = s
		print("Destroyed Ship: " + d_name.split("_")[0] + " xLoc: " + str(d_x))
		ships.remove(s)
		# step ships
		for i2 in range(0, len(ships)):
			(name, x, y) = ships[i2]
			s_type = name.split("_")[1]
			if s_type == "A":
				if x > 0:
					x -= 10
				else:
					x += 10
			elif s_type == "B":
				if x > 0:
					x -= 20
				else:
					x += 20
			elif s_type == "C":
				if x > 0:
					x -= 30
				else:
					x += 30
			ships[i2] = (name, x, y)
			
